tests/validation/src/jacobian_validation.cpp

```diff
@@ -0,0 +1,1475 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s)
+#include "detray/definitions/units.hpp"
+#include "detray/detectors/bfield.hpp"
+#include "detray/detectors/create_telescope_detector.hpp"
+#include "detray/propagator/actors/parameter_resetter.hpp"
+#include "detray/propagator/actors/parameter_transporter.hpp"
+#include "detray/propagator/navigator.hpp"
+#include "detray/propagator/propagator.hpp"
+#include "detray/propagator/rk_stepper.hpp"
+#include "detray/simulation/event_generator/track_generators.hpp"
+#include "detray/utils/inspectors.hpp"
+#include "detray/utils/statistics.hpp"
+#include "tests/common/tools/intersectors/helix_line_intersector.hpp"
+#include "tests/common/tools/intersectors/helix_plane_intersector.hpp"
+
+// Vecmem include(s)
+#include <vecmem/memory/host_memory_resource.hpp>
+
+// GTest include(s)
+#include <gtest/gtest.h>
+
+// Boost
+#include <boost/program_options.hpp>
+
+// System include(s).
+#include <filesystem>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+
+namespace po = boost::program_options;
+using namespace detray;
+
+// Type declarations
+using transform3_type = __plugin::transform3<scalar>;
+using vector3 = __plugin::vector3<scalar>;
+using bound_vector_type = bound_track_parameters<transform3_type>::vector_type;
+using bound_covariance_type =
+    bound_track_parameters<transform3_type>::covariance_type;
+using matrix_operator = typename transform3_type::matrix_actor;
+using size_type = typename transform3_type::size_type;
+template <size_type ROWS, size_type COLS>
+using matrix_type = typename matrix_operator::template matrix_type<ROWS, COLS>;
+
+namespace {
+
+// Setup
+const vector3 B_z{0.f, 0.f, 1.997f * unit<scalar>::T};
+const std::array<scalar, 5u> h_sizes_rect = {1e-1f, 1e-1f, 2e-2f, 2e-2f, 1e-2f};
+const std::array<scalar, 5u> h_sizes_wire = {1e-1f, 1e-1f, 2e-2f, 2e-2f, 1e-2f};
+const scalar constraint_step_size = 10000.f * unit<scalar>::mm;
+
+std::random_device rd;
+std::mt19937 mt(rd());
+
+// Detector length generator
+constexpr const scalar min_length = 100.f * unit<scalar>::mm;
+constexpr const scalar max_length = 700.f * unit<scalar>::mm;
+std::uniform_real_distribution<scalar> rand_length(min_length, max_length);
+// Wiggling the Destination surface
+std::uniform_real_distribution<scalar> rand_tilt(-constant<scalar>::pi / 4.f,
+                                                 constant<scalar>::pi / 4.f);
```

Comment:
Not terribly important, but:
```suggestion
std::uniform_real_distribution<scalar> rand_tilt(-constant<scalar>::pi_4,
                        constant<scalar>::pi_4);
```

Reply:
Fixed